src/DebuggingFeature.ts

```diff
@@ -0,0 +1,80 @@
+import { BaseLanguageClient, StaticFeature, Neovim } from "coc.nvim";
+import {
+  ExecuteCommandParams,
+  ExecuteCommandRequest,
+  InitializeParams,
+  ServerCapabilities
+} from "vscode-languageserver-protocol";
+import { commands } from "coc.nvim";
+
+interface DebugAdapterStartResponse {
+  name: string;
+  uri: string;
+}
+
+export class DebuggingFeature implements StaticFeature {
+  private withVimspector = false;
+
+  constructor(private _nvim: Neovim, private _client: BaseLanguageClient) {}
+
+  async preInit(): Promise<void> {
+    this.withVimspector = (await this._nvim.getVar("loaded_vimpector")) === 1;
```

Comment:
`g:loaded_vimspector` is not strictly a public API, but I think the convention is probably sound enough.

At what point in vim startup does this code run ? Vimspector may be loaded by a `:packadd` command otherwise lazily loaded, so it might not be correct to only run this check once (for example)

Reply:
The timing _should_ be ok, since this will first be ran once the project is recognized as a Scala project. However, as a follow-up to this comment, are we able to check in a different way to just ensure that vimspector is installed, and then do this check later when we are about to actually initialize the process?